bundle/src/main/java/io/wcm/caconfig/editor/TagBrowserRootPathProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * #%L
+ * wcm.io
+ * %%
+ * Copyright (C) 2022 wcm.io
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package io.wcm.caconfig.editor;
+
+import io.wcm.caconfig.editor.EditorProperties;
+import org.apache.sling.api.resource.Resource;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ConsumerType;
+
+/**
+ * Allows to provide a dynamic root path for tag browser widget.
+ * Implement as OSGi service and set a service property {@link #PROPERTY_SELECTOR} which is used to reference
+ * the implementation in the context-aware configuration property metadata options.
+ */
+@ConsumerType
+public interface TagBrowserRootPathProvider {
```

Comment:
is this necessary? Isnt it the same as the PathBrowserRootPathProvider (same for the unit tests that accompany it)

Reply:
Yes but I copied it to allow setting a tagbrowserRootPath/tagbrowserRootPathContext/tagbrowserRootPathProvider so you wouldn't have to use **pathbrowser**RootPath in a tagbrowser config.